src/test/java/seedu/address/logic/commands/DeleteGoodCommandTest.java

```diff
@@ -0,0 +1,112 @@
+package seedu.address.logic.commands;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandFailure;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.address.logic.commands.CommandTestUtil.showGoodAtIndex;
+import static seedu.address.testutil.TypicalGoods.APPLE;
+import static seedu.address.testutil.TypicalGoods.getTypicalInventory;
+import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_GOOD;
+import static seedu.address.testutil.TypicalIndexes.INDEX_SECOND_GOOD;
+import static seedu.address.testutil.TypicalSuppliers.getTypicalAddressBook;
+import static seedu.address.testutil.TypicalTransactions.getTypicalTransactionHistory;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.good.Good;
+
+class DeleteGoodCommandTest {
+
+    private Model model = new ModelManager(getTypicalAddressBook(), getTypicalInventory(),
+            getTypicalTransactionHistory(), new UserPrefs());
+
+    @Test
+    public void execute_validIndexUnfilteredList_success() {
+        Good goodToDelete = model.getFilteredGoodList().get(INDEX_FIRST_GOOD.getZeroBased());
+        DeleteGoodCommand deleteGoodCommand = new DeleteGoodCommand(INDEX_FIRST_GOOD);
+
+        String expectedMessage = String.format(DeleteGoodCommand.MESSAGE_DELETE_GOOD_SUCCESS,
+                goodToDelete.getGoodName().fullGoodName);
+
+        ModelManager expectedModel = new ModelManager(model.getAddressBook(), getTypicalInventory(),
+                getTypicalTransactionHistory(), new UserPrefs());
```

Comment:
Why call model.getAddressBook but call getTypical for everything else?

Reply:
Standardized.

Reply:
In this case, there is some changes in the inventory due to delete good. Then it should be 
new ModelManager(getTypicalAddressBook(), model.getInventory(),
                getTypicalTransactionHistory(), new UserPrefs())
since the inventory has changed.
please adjust this for all cases in this class.

Reply:
Mutating transaction history is the next delete command enhancement planned after this initial implementation.